Zend/zend_compile.c

```diff
@@ -9501,7 +9501,16 @@ static void zend_compile_encaps_list(znode *result, zend_ast *ast) /* {{{ */
 	j = 0;
 	last_const_node.op_type = IS_UNUSED;
 	for (i = 0; i < list->children; i++) {
-		zend_compile_expr(&elem_node, list->child[i]);
+		zend_ast *encaps_var = list->child[i];
+		if (encaps_var->attr & (ZEND_ENCAPS_VAR_DOLLAR_CURLY|ZEND_ENCAPS_VAR_DOLLAR_CURLY_VAR_VAR)) {
+			if (encaps_var->attr & ZEND_ENCAPS_VAR_DOLLAR_CURLY_VAR_VAR) {
+				zend_error(E_DEPRECATED, "Using ${} (variable variables) in strings is deprecated");
+			} else {
+				zend_error(E_DEPRECATED, "Using ${} in strings is deprecated");
```

Comment:
```suggestion
				zend_error(E_DEPRECATED, "Using ${var} in strings is deprecated, use {$var} instead");
```
Maybe also add what should be used instead. It's not like variable substitutions in general are deprecated, just the ${} syntax.

Reply:
Nice suggestion, thanks :slightly_smiling_face: